### DISCUSSION TITLE
Using cache handler with pages file which uses getServersideProps

### DISCUSSION THREAD

**Discussion by sreekanthjnayar** [vercel/next.js#65661] [Help]
### Summary

Is it possible to use the cache handler to cache the content of a pages file rendered using get server side props. Our application involves rendering pages where the content may vary by URL. We do use CDN to ensure the load on the server is minimal. Before moving to the folder driven architecture of nextjs we had implemented a custom express server where we use to get the rendered HTML using renderToHTML and cache the HTML in redis. With the new folder driven approach, is there a way to do this withiout having to revert to use a custom express server.

**Comment by SouthLink** [comment]
You may need to [configuration-caching](https://nextjs.org/docs/pages/building-your-application/deploying#configuring-caching)

**Reply by sreekanthjnayar** [reply]
I had tried setting up a cache handler like the way it is shown in the documentation. When my route is invoked (which uses getServersideProps to fetch the data), I could see that the constructor was getting invoked while the other methods were not. Hence I was not able to take control of the rendering. Could you advise how to use cache handler when we have getServersideProps used.

**Reply by SouthLink** [reply]
Whether you have to debug your cache program during the production run after the build, the local environment tuning may not take effect, You can refer to this [issue](https://github.com/vercel/next.js/issues/58094)

**Reply by sreekanthjnayar** [reply]
Thank you, will give this a shot

**Reply by sreekanthjnayar** [reply]
I tried setting the NODE_ENV as production, added console.log statements to see if this works (with the pages where we use getServerSideProps). It doesn't seem to work still. I saw a comment in the link mentioned in the earlier comment which states this _**"The cache is used for SSG/ISR pages in Pages and App Router, fetch call results, and [unstable_cache](https://nextjs.org/docs/app/api-reference/functions/unstable_cache) callback calls in App Router. Server-side rendered pages, in general, are not affected by the cache in any way — such pages are marked as dynamic. Next.js doesn't cache them"**_ . Is this the reason why it doesn't work; which means you cannot use a cache manager for a page which uses getServerSideProps.

**Reply by SouthLink** [reply]
This is the function of the app directory, which should have nothing to do with your question. Are you using the next build program to test the cache function

**Reply by sreekanthjnayar** [reply]
Yes, I am using next build with node env set as production. I don't see the get or set function in the cache handler being called when I hit a route (which uses pages with get server side props).

**Reply by SouthLink** [reply]
Is your next.js updated to the latest version, and if it still doesn't work, you may want to provide a minimal reproduction example to the next issues

**Comment by sreekanthjnayar** [comment]
I dug a bit more into the nextjs caching and took a look at the following.
https://nextjs.org/docs/app/api-reference/functions/unstable_cache
https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#fetching-data-on-the-client-with-route-handlers

It is pretty clear to me that when dynamic rendering (using getServerSideProps) is involved there is no built in caching mechanism and none of the above caching mechanism can be used as well. The solution that I have finally decided to go with is to have a simple expressjs application above our main app which acts a proxy as well as takes care of caching the rendered HTML returned by the main app.
